src/cdef.rs

```diff
@@ -549,3 +556,703 @@ pub unsafe fn fill(mut tmp: *mut i16, stride: ptrdiff_t, w: c_int, h: c_int) {
+                return;
+            }
+
+            (*c).dir = dav1d_cdef_dir_8bpc_sse4;
+            (*c).fb[0] = dav1d_cdef_filter_8x8_8bpc_sse4;
+            (*c).fb[1] = dav1d_cdef_filter_4x8_8bpc_sse4;
+            (*c).fb[2] = dav1d_cdef_filter_4x4_8bpc_sse4;
+
+            #[cfg(target_arch = "x86_64")]
+            {
+                if !flags.contains(CpuFlags::AVX2) {
+                    return;
+                }
+
+                (*c).dir = dav1d_cdef_dir_8bpc_avx2;
+                (*c).fb[0] = dav1d_cdef_filter_8x8_8bpc_avx2;
+                (*c).fb[1] = dav1d_cdef_filter_4x8_8bpc_avx2;
+                (*c).fb[2] = dav1d_cdef_filter_4x4_8bpc_avx2;
+
+                if !flags.contains(CpuFlags::AVX512ICL) {
+                    return;
+                }
+
+                (*c).fb[0] = dav1d_cdef_filter_8x8_8bpc_avx512icl;
+                (*c).fb[1] = dav1d_cdef_filter_4x8_8bpc_avx512icl;
+                (*c).fb[2] = dav1d_cdef_filter_4x4_8bpc_avx512icl;
+            }
+        }
+        BPC::BPC16 => {
+            if !flags.contains(CpuFlags::SSSE3) {
+                return;
+            }
+
+            (*c).dir = dav1d_cdef_dir_16bpc_ssse3;
+            (*c).fb[0] = dav1d_cdef_filter_8x8_16bpc_ssse3;
+            (*c).fb[1] = dav1d_cdef_filter_4x8_16bpc_ssse3;
+            (*c).fb[2] = dav1d_cdef_filter_4x4_16bpc_ssse3;
+
+            if !flags.contains(CpuFlags::SSE41) {
+                return;
+            }
+
+            (*c).dir = dav1d_cdef_dir_16bpc_sse4;
+
+            #[cfg(target_arch = "x86_64")]
+            {
+                if !flags.contains(CpuFlags::AVX2) {
+                    return;
+                }
+
+                (*c).dir = dav1d_cdef_dir_16bpc_avx2;
+                (*c).fb[0] = dav1d_cdef_filter_8x8_16bpc_avx2;
+                (*c).fb[1] = dav1d_cdef_filter_4x8_16bpc_avx2;
+                (*c).fb[2] = dav1d_cdef_filter_4x4_16bpc_avx2;
+
+                if !flags.contains(CpuFlags::AVX512ICL) {
+                    return;
+                }
+
+                (*c).fb[0] = dav1d_cdef_filter_8x8_16bpc_avx512icl;
+                (*c).fb[1] = dav1d_cdef_filter_4x8_16bpc_avx512icl;
+                (*c).fb[2] = dav1d_cdef_filter_4x4_16bpc_avx512icl;
+            }
+        }
+    };
+}
+
+#[inline(always)]
+#[cfg(all(feature = "asm", any(target_arch = "arm", target_arch = "aarch64"),))]
+unsafe extern "C" fn cdef_filter_8x8_neon_erased(
+    dst: *mut DynPixel,
+    stride: ptrdiff_t,
+    left: *const LeftPixelRow2px<DynPixel>,
+    top: *const DynPixel,
+    bottom: *const DynPixel,
+    pri_strength: c_int,
+    sec_strength: c_int,
+    dir: c_int,
+    damping: c_int,
+    edges: CdefEdgeFlags,
+    _bitdepth_max: c_int,
+) {
+    use crate::src::align::Align16;
+
+    let mut tmp_buf = Align16([0; 200]);
+    let tmp = tmp_buf.0.as_mut_ptr().offset(2 * 16).offset(8);
+    dav1d_cdef_padding8_8bpc_neon(tmp, dst, stride, left, top, bottom, 8, edges);
+    dav1d_cdef_filter8_8bpc_neon(
```

Comment:
You're calling the bitdepth-specific versions here; this `fn` needs to be generic and dispatch on `BD::BPC`.